### ISSUE TITLE
what does value for fetching content represent?

### ISSUE THREAD

**Issue by SethDocherty** [easierdata/ipfs-stac#40]
When fetching content from IPFS, what does the number represent? 
`⠙ Fetching bafybeiddydzzb5z2smwos72qzci3wchhk4kk42i23jhfeldhkczsusq5ra - 0/33236 `

I'm guessing it may be related to the total size of the file based on running `ipfs dag stat`

```
CID                                                             Blocks          Size
bafybeiddydzzb5z2smwos72qzci3wchhk4kk42i23jhfeldhkczsusq5ra     318             332378721

Summary
Total Size: 332378721
Unique Blocks: 318
Shared Size: 0
Ratio: 1.000000
```

Making the number human-readable with a unit of MB would be ideal.

**Comment by SethDocherty** [comment]
I'm noticing that in Jupyter Notebook's, the part of the string representing the **file size** is not updated during each loop. Let's also convert bytes to MB as display units.    

@matthewnanas, could you also look into this warning for Jupyter Notebook?

![image](https://github.com/user-attachments/assets/759d6081-03ef-4c1a-bd99-cec96f50f3ce)

**Comment by matthewnanas** [comment]
@SethDocherty that error should be fixed with https://github.com/easierdata/ipfs-stac/commit/92365e2b39adbdeafdafbd74b68b2d5d10188a94 , check it out and let me know if you still encounter the same problem.

**Comment by SethDocherty** [comment]
@matthewnanas here's a snippet of how to remove the warning for jupyter notebook and how to change the property to `None`.

```Python
# Disable yaspin/core.py:128: UserWarning: color, on_color and attrs are not supported when running in jupyter notebook
# Check if running in Jupyter Notebook
try:
    get_ipython()
    yaspin._color = None
except Exception as e:
    pass
```

I'm not sure of the best way to handle this is to implement right in the `Client` or within juypyter notebook.

I think this also supresses all warning from appearing in a notebook, so it would supress the yaspin warning + any other warnings that may appear which may not be ideal

`warnings.filterwarnings("ignore", category=UserWarning)`

**Comment by matthewnanas** [comment]
@SethDocherty Sorry I misinterpreted your previous comment, the latest commit should render the progress in MB now. I think it makes more sense to handle suppressing warnings in the notebook itself. I just perceive them to be sandbox environments that could ultimately be published. Let me know your thoughts.

**Comment by SethDocherty** [comment]
This has been resolved in PR #45

**Comment by SethDocherty** [comment]
If we would like to make the file size properties dynamic, we can use something like this:

```python
class DynamicProgressText:
    def __init__(self, cid, total_size):
        self.cid = cid
        self.progress = 0
        self.total_size_bytes = total_size
        self.total_size = self.format_size(total_size)
        self._start = datetime.now()

    def update_progress(self, progress):
        self.progress = progress

    def format_size(self, size_in_bytes):
        units = ["B", "KB", "MB", "GB", "TB"]
        unit_index = 0

        while size_in_bytes >= 1024 and unit_index < len(units) - 1:
            size_in_bytes /= 1024
            unit_index += 1

        return f"{size_in_bytes:.4f} {units[unit_index]}"

    def __str__(self):
        now = datetime.now()
        delta = now - self._start
        cid_short = self.cid.split("/")[-1]
        progress_text = self.format_size(self.progress)
        return f"Fetching {cid_short} - {progress_text}/{self.total_size} ({round(delta.total_seconds(), 1)}s)"
```
